saltproc/tests/test_depcode.py

```diff
@@ -163,30 +197,166 @@ def test_replace_burnup_parameters():
+    ----------
+    object1 : openmc.Surface, openmc.Universe, openmc.Cell, openmc.Material, \
+    openmc.Lattice
+        First openmc object to compare
+    object2 :  openmc.Surface, openmc.Universe, openmc.Cell, openmc.Material, \
+    openmc.Lattice
+        Second openmc object to compare
+
+    """
+    try:
+        object_type = type(object1)
+        assert isinstance(object2, object_type)
+        assert object1.id == object2.id
+        # this may not apply to all objects.
+        assert object1.name == object2.name
+        # need to check
+        if object_type == om.Material:
+            assert object1.density == object2.density
+            assert object1.nuclides == object2.nuclides
+            assert object1.temperature == object2.temperature
+            assert object1.volume == object2.volume
+            assert object1._sab == object2._sab
+
+        elif object_type == om.Cell:
+            assert object1.fill_type == object2.fill_type
+            _check_none_or_openmc_object_equal(object1.fill, object2.fill)
+            assert object1.region == object2.region
+            _check_none_or_iterable_of_ndarray_equal(
+                object1.rotation, object2.rotation)
+            _check_none_or_iterable_of_ndarray_equal(
+                object1.rotation_matrix, object2.rotation_matrix)
+            _check_none_or_iterable_of_ndarray_equal(
+                object1.translation, object2.translation)
+            assert object1.volume == object2.volume
+            # assert object1.atoms == object2.atoms
+
+        elif issubclass(object_type, om.Lattice):
+            assert object1.shape == object2.shape
+            assert object1.lower_left == object2.lower_left
+            assert object1.pitch == object2.pitch
+            _check_none_or_openmc_object_equal(object1.outer, object2.outer)
+            _check_openmc_iterables_equal(
+                {'univ': (object1.universes, object2.universes)})
+
+        elif issubclass(object_type, om.Surface):
+            assert object1.boundary_type == object2.boundary_type
+            _check_none_or_iterable_of_ndarray_equal(
+                object1.coefficients, object2.coefficients)
+            assert object1.type == object2.type
+
+        elif object_type == om.Universe:
+            _check_openmc_iterables_equal(
+                {'cells': (object1.cells, object2.cells)})
+            assert object1.volume == object2.volume
+            _check_none_or_iterable_of_ndarray_equal(
+                object1.bounding_box, object2.bounding_box)
+        else:
+            raise ValueError(
+                f"Object of type {object_type} is not an openmc object.")
+
+    except AssertionError:
+        raise AssertionError(
+            f"objects of type {object_type} with ids {object1.id} and \
+            {object2.id} not equal")
+
+
+def _check_openmc_iterables_equal(iterable_dict):
+    """
+    Helper function to check if the given dictionary of iterbales
+
```

Comment:
It is not clear what is being tested here.

Reply:
I updated the docstring to be a bit clearer